### PR TITLE
Tweak output of gen_changelog.py

### PR DESCRIPTION
**Pull request by mwoehlke-kitware**
Adjust the output of `gen_changelog.py` so that it is easier to convert to the desired final format. I think this strikes a good balance between minimal additional tweaking needed while still usefully providing the information that helps a human sort out things like renamed or transient options that need manual tweaking.

### CODE REVIEW

**Comment by mwoehlke-kitware** [comment]
Updated example output:
```
$ ./scripts/gen_changelog.py uncrustify-0.69.0
710b9b90d063e1fac853b77ad9509bed37ff31ee
  Added   : mod_sort_case_sensitive              May 12 2019
5d01af29d1c99bed11e3e2b996dddcf6d2de541b
  Added   : sp_trailing_ret_t                    Jul 12 2019
b8e67e92bc0066f29e6e18574c93817dd33a78ea
  Added   : sp_before_square_ignore_asm_block    Jul 23 2019
40c028454904f4fc87bd93b9703eaeb58fa57dd0
  Added   : sp_before_square_asm_block           Jul 29 2019
  Removed : sp_before_square_ignore_asm_block    Jul 29 2019
b9b7f2fffb2c73925906ca7e5c8536d03c93c4bc
  Added   : nl_multi_line_sparen_open            Aug  3 2019
  Added   : nl_multi_line_sparen_close           Aug  3 2019
83a014197001ea815f92d977fc30aa36dec39e1b
  Added   : sp_cpp_lambda_square_paren           Aug 20 2019
  Added   : sp_cpp_lambda_square_brace           Aug 20 2019
  Added   : sp_cpp_lambda_paren_brace            Aug 20 2019
  Removed : sp_cpp_lambda_paren                  Aug 20 2019
8b6455c08c4f70b04b03b48c32fa14b9014354b4
  Added   : use_sp_after_angle_always            Sep 13 2019
264ec3c988b5667c971b0d6508b17d0c9337c82c
  Added   : sp_cpp_lambda_fparen                 Sep 13 2019
883ccefc3bf8c8cdf6b159f280e22c2d72ce83f5
  Added   : sp_func_type_paren                   Oct 11 2019
8284db41edbb777829f26d247470c75d3d317641
  Added   : nl_before_namespace                  Oct  8 2019
  Added   : nl_after_namespace                   Oct  8 2019
3af190ceda3f8025028b5f8aae7541d35bda39c7
  Added   : nl_template_class_def                Oct  2 2019
  Added   : nl_template_func_def_special         Oct  2 2019
  Added   : nl_template_using                    Oct  2 2019
  Added   : nl_template_func_def                 Oct  2 2019
  Added   : nl_template_class_decl               Oct  2 2019
  Added   : nl_template_var                      Oct  2 2019
  Added   : nl_template_class_decl_special       Oct  2 2019
  Added   : nl_template_func                     Oct  2 2019
  Added   : nl_template_func_decl_special        Oct  2 2019
  Added   : nl_template_func_decl                Oct  2 2019
  Added   : nl_template_class_def_special        Oct  2 2019
e9bc08c3baa3b7545a592b5e1d0d23e530c58b7f
  Added   : sp_trailing_return                   Oct 23 2019
  Removed : sp_trailing_ret_t                    Oct 23 2019
d783ecfe7f08b525eb882ff3a7c6525d8866d54b
  Added   : nl_oc_before_interface               Oct 28 2019
  Added   : nl_oc_before_implementation          Oct 28 2019
  Added   : nl_oc_before_end                     Oct 28 2019
```

**Comment by mwoehlke-kitware** [comment]
...and the above is a decent example of why I don't think it makes sense to try to use the script's output directly:
- `sp_trailing_ret_t` is a "transient" option; that is, it was both added and removed since the last release. (In this case, the removal is a case of changing the name before release; the final name is `sp_trailing_return`.)
  - We might be able to filter transients, but I'm not 100% confident that is always the right thing to do, and they should be sufficiently uncommon that I'm not sure the benefit is worth the risk.
- `sp_cpp_lambda_paren` was renamed to `sp_cpp_lambda_square_paren`.
  - Although this was a "simple" rename, there are other new options in this commit, and it is not feasible for the script to determine which is the new name, and which are just new options. Moreover, there is no assurance that a paired removal and addition will be such a "simple" rename. It's arguable in this case if the new name should be listed as a "new" option.
- `sp_before_square_ignore_asm_block` was superseded by `sp_before_square_asm_block`.
  - In this case, a completely new feature made an old option obsolete. There is no question that the new option is "new", and should be listed as such. As with the previous point, there is no feasible way for the script to know the difference.

In short, while I am fairly confident the script has no false negatives (i.e. will always correctly identify "simple" instances of new option), it will have false positives, and any detected removal likely needs human intervention to decide what happened. (False positives *should*, at least, always correspond to a removal.)
